job-runner.toml

```diff
@@ -0,0 +1,37 @@
+# Install to ~/.config/cockpit-dev/job-runner.toml
+
+[container]
+command = 'podman'
+args = []
+run-args = ['--device=/dev/kvm']
+default-image = 'quay.io/cockpit/tasks:latest'
+
+[container.secrets]
+# see podman-secret(1)
+# github-token = '--secret=github-token'
+
+[logs]
+driver='local'   # 's3' or 'local'
+
+[logs.s3]
+# hint: podman run --rm --net=host quay.io/minio/minio server /var
+url = 'http://127.0.0.1:9000/tmp/'
+key = {access='minioadmin', secret='minioadmin'}
+user-agent = 'job-runner (cockpit-project/bots)'
+acl = 'public-read'
+
+[logs.local]
+# hint: python -m http.server -b 127.0.0.1 -d ~/.cache/cockpit-dev/job-runner-logs
+dir = '~/.cache/cockpit-dev/job-runner-logs'
+link = 'http://127.0.0.1:8000/'
+
+[forge]
+driver='github'
+
+[forge.github]
+clone-url = 'https://github.com/'
+api-url = 'https://api.github.com/'
+post = true  # whether to post statuses, open issues, etc.
+user-agent = 'job-runner (cockpit-project/bots)'
+# (at least) one of `token` or `post = false` must be set
+# token = 'ghp_XXX'
```

Comment:
That feels redundant. job-runner itself should just default to `${COCKPIT_GITHUB_TOKEN_FILE:-~/.config/cockpit-dev/github-token}`, in other words, use the `task.github` API. And the spawned temp containers would use `container.secrets.github-token`. Why is this needed here *again*?

Reply:
This is worth discussing, at least as a fallback.  Ditto for S3 keys.  Let's keep that in mind for a follow-up.

Reply:
Tracked in #6030 now. However, is that actually hard to fix? It's already commented out after all, and you said that this is the internal defaults -- so doesn't that mean that there is no default, and it somehow works without anyway?

If it's hard, then let's do it as a follow-up, but I'd like to at least understand why.

Reply:
It's absolutely not difficult to do, at least for the GitHub token.  The S3 stuff is slightly more complicated since we'd have to bring back the dir-scanning/suffix-matching logic.

For me it's more a question of the reasonableness of the default.  Do you picture a situation where we should use job-runner locally on real PRs and have statuses posted from our own GitHub accounts?  That's effectively what you're requesting here.

I think the "broken by default" default config is a good thing...

Reply:
I was missing that you *intentionally* wanted to "break" the default. So this stops being a blocker for this PR for me, and we can continue to discuss it in #6030.

My worry is more that the intended mode of operation requires duplicating my github token, it has to be both in ~/.config and in that toml. That makes rotating tokens harder, and I like "single source of truth".

Reply:
For the reason above, *if* we wanted to solve the broken-by-default thing, I'd be fairly insistent about solving it in the `forge.github.post = false` direction.